catalog/java/io/material/catalog/lists/MaterialListAdapterDemoFragment.java

```diff
@@ -0,0 +1,117 @@
+public class MaterialListAdapterDemoFragment extends DemoFragment {
+
+  @Override
+  @SuppressWarnings("RestrictTo")
+  public View onCreateDemoView(
+      LayoutInflater layoutInflater, @Nullable ViewGroup viewGroup, @Nullable Bundle bundle) {
+
+    View layout = layoutInflater.inflate(R.layout.cat_material_list_adapter_fragment, viewGroup, false);
+
+    RecyclerView recyclerView = layout.findViewById(R.id.recycler_view);
+    recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+    recyclerView.setAdapter(materialListAdapters[0]);
+
+    final List<String> options = Arrays.asList("Single Line Item", "Two Line Item", "Three Line Item");
+
+    AutoCompleteTextView editTextFilledExposedDropdown =
+        layout.findViewById(R.id.filled_exposed_dropdown);
+    editTextFilledExposedDropdown.setText(options.get(0));
+    editTextFilledExposedDropdown.setAdapter(new ArrayAdapter<>(
+        getContext(),
+        R.layout.cat_popup_item,
+        options));
+    editTextFilledExposedDropdown.addTextChangedListener(new TextWatcher() {
+      @Override
+      public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+      }
+
+      @Override
+      public void onTextChanged(CharSequence s, int start, int before, int count) {
+      }
+
+      @Override
+      public void afterTextChanged(Editable s) {
+        editTextFilledExposedDropdown.setAdapter(new ArrayAdapter<>(
+            getContext(),
+            R.layout.cat_popup_item,
+            options));
+        recyclerView.setAdapter(materialListAdapters[options.indexOf(s.toString())]);
+      }
+    });
+
+    return layout;
+  }
+
+  MaterialListAdapter[] materialListAdapters = {
+      new SingleLineItemAdapter(1000, new MaterialListBinder<SingleLineItemViewHolder>() {
+        @Override
+        public void onBind(SingleLineItemViewHolder vh, int position) {
+          vh.text.setText(R.string.mtrl_list_item_one_line);
+          vh.icon.setImageResource(R.drawable.logo_avatar_anonymous_40dp);
+        }
+      }),
+      new TwoLineItemAdapter(1000, new MaterialListBinder<TwoLineItemViewHolder>() {
+        @Override
+        public void onBind(TwoLineItemViewHolder vh, int position) {
+          vh.primaryText.setText(R.string.mtrl_list_item_two_line);
+          vh.secondaryText.setText(R.string.mtrl_list_item_secondary_text);
+          vh.icon.setImageResource(R.drawable.logo_avatar_anonymous_40dp);
+        }
+      }),
+      new ThreeLineItemAdapter(1000, new MaterialListBinder<ThreeLineItemViewHolder>() {
+        @Override
+        public void onBind(ThreeLineItemViewHolder vh, int position) {
+          vh.primaryText.setText(R.string.mtrl_list_item_one_line);
+          vh.secondaryText.setText(R.string.mtrl_list_item_secondary_text);
+          vh.tertiaryText.setText(R.string.mtrl_list_item_tertiary_text);
+          vh.icon.setImageResource(R.drawable.logo_avatar_anonymous_40dp);
+        }
+      })};
```

Comment:
The dataset is irrelevant, this demo is meant to show that the relationship between MaterialListBinder and MaterialListAdapter works

Reply:
I like this demo because it shows the different list types that exist, so I think it's good to keep, but maybe you could make another demo that shows how someone might actually use `MaterialListBinder` to bind data. A simple case, like binding an array of strings or an array of simple objects, would be useful.